src/gui/QvisColorTableWindow.C

```diff
@@ -810,15 +1053,30 @@ QvisColorTableWindow::UpdateNames()
         }
         // Set the text of the default color table into the name line edit.
         nameLineEdit->setText(QString(colorAtts->GetNames()[index].c_str()));
-        categoryLineEdit->setText(QString(colorAtts->GetColorTables(index).GetCategoryName().c_str()));
+        tagLineEdit->setText(QString(colorAtts->GetColorTables(index).GetTagsAsString().c_str()));
     }
 
+    tagTable->blockSignals(false);
     nameListBox->blockSignals(false);
     defaultContinuous->blockSignals(false);
     defaultDiscrete->blockSignals(false);
 
     // Set the enabled state of the delete button.
     deleteButton->setEnabled(colorAtts->GetNumColorTables() > 1);
+
+    static bool run_before = false;
+    if (!run_before)
+    {
+        // This only needs to happen the very first time for loading options.
```

Comment:
I find this comment worrisome. It isn't clear that the approach taken (a static bool) was chosen because this only ever has to happen once (the first time) vs. this was the only way to prevent the code from crashing. Is this hiding something potentially more insidious?

Reply:
Well it's a bit of both; the code in this block does need to happen so that the filtered down list of color tables appears in all the buttons. It also only ever needs to happen once, because of the order of execution of functions when loading saved config/session files. Before this band-aid, the buttons didn't get updated with my filtering selection until you clicked something in the color table manager, which is a bug. So it definitely needs to happen, and only needs to happen once.

However, if it happens on every call, VisIt will not only freeze, but I get booted off my compute node. There is an underlying issue that I know about that caused me to add this band-aid here as well as the band-aid you flagged that had my `this is very scary` comment. I'm going to open an issue for this at some point today, and get to work on that after. There might not be a better way around this problem, but I'll investigate further.